### ISSUE TITLE
"Show source code in..." should be more subtle

### ISSUE THREAD

**Issue by cs01** [pawamoy/mkdocstrings#35]
When viewing documentation, I rarely/never want to look at the underlying source code. Assuming the majority of use cases is to not view source, the visual real estate for the "Show source..." button should be smaller.

**Comment by pawamoy** [comment]
What would you suggest as to replace the "Show source..." admonition 🙂 ?

Something lighter would be a simple summary/details HTML thing: without any admonition class it's a single text line with a small expanded/collapsed triangle icon on the left.

I'm also planning to make "Show source code..." optional with config options (globally or per autodoc instruction).

**Comment by cs01** [comment]
> What would you suggest as to replace the "Show source..." admonition slightly_smiling_face ?

Small text with a triangle next to it saying "Show Source". An example of this is in black's readme. Search for "A compatible `.isort.cfg`" in https://github.com/psf/black#how-black-wraps-lines.

One other thing I ran into is when searching for text in the rendered UI, the match preview shows a bunch of numbers next to the matched function name because it's showing the number of the source code. Ideally the numbers would be hidden from the search results, but if that can't be done, moving the source to the bottom of the doc block would probably result in better search previews.

**Comment by pawamoy** [comment]
Indeed, just tested with your https://cs01.github.io/pygdbmi/api/gdbmiparser docs.

I'll try to put the source code at the end to see if it fixes the search preview.

Maybe we can add a configuration option to show or hide line numbers in "Show source code" blocks as well.

And maybe there's room for improvement in mkdocs-search plugin itself, to hide line numbers since I guess nobody is interested in them showing up in the search preview.

**Comment by pawamoy** [comment]
The new architecture is almost ready. I used the chance to improve the rendering. I moved the "Show source code" block under the docstring, so line numbers don't appear in the search. I've also used a "quote" class instead of nothing (which defaulted to "note"), so instead of a blue admonition, we have a gray one, a bit less eye-catching :slightly_smiling_face: 

Also note that you'll be able to override the templates!

I'd like to really resolve the line numbers showing up in search issue, so I'll leave this one open.

**Comment by pawamoy** [comment]
New version (0.9.1) is published, let me know if you have any problems with it.

**Comment by pawamoy** [comment]
I believe the current state is sufficient (optional code blocks, end of the docstring, not too eye-catching).
Also, the "line numbers in search result" issue seems to be fixable by using [the `pymdownx-inline` style](https://github.com/mkdocs/mkdocs/issues/1965#issuecomment-586766725).

**Comment by ssbarnea** [comment]
@pawamoy I do not think that something that takes around 30% of the entire page can be called subtle.

![](https://sbarnea.com/ss/Screen-Shot-2023-01-24-15-12-19.91.png)

What I would call subtle would be more towards having a simple `[src]` dimmed text after the member/method name, something that would not take any horizontal space on the page.

Don't get me wrong, the "show source" is an extremely useful feature. Is just that I wish it would be less-in-your-face.

**Comment by pawamoy** [comment]
You're right, we can do better. Can you open a new feature request?
